pages/apim/3.x/event-native-apim/event-native-apim-documentation.adoc

```diff
@@ -0,0 +1,145 @@
+
+The following entrypoints are available.
+
+* SSE
+* Websockets
+* HTTP GET
+* HTTP POST
+
+The following endpoints are available
+
+* Kafka
+* MQQT
+
+== Create a v4 async API 
+
+The link:https://github.com/gravitee-io/postman-collections[Gravitee Postman Collections] contain several examples of how to create an API.
+
+For example, to create an HTTP POST entrypoint that connects to a Kafka endpoint, send a POST request to `{\{management_host}}/management/organizations/DEFAULT/environments/DEFAULT/v4/apis/`, where `{\{management_host}}` is the host for the management API, with the following message body:
+
+[source json]
+----
+{
+    "name": "Data Ingestion to Kafka",
+    "apiVersion": "1.0",
+    "definitionVersion": "4.0.0",
+    "type": "async",
+    "description": "Data Ingestion to Kafka",
+    "listeners": [
+        {
+            "type": "http",
+            "paths": [
+                {
+                    "path": "/data/ingestion/kafka"
+                }
+            ],
+            "entrypoints": [
+                {
+                    "type": "http-post",
+                    "configuration": {
+                        "requestHeadersToMessage": false
+                    }
+                }
+            ]
+        }
+    ],
+    "endpointGroups": [
+        {
+            "name": "default",
+            "type": "kafka",
+            "endpoints": [
+                {
+                    "name": "default",
+                    "type": "kafka",
+                    "weight": 1,
+                    "inheritConfiguration": false,
+                    "configuration": {
+                        "bootstrapServers": "kafka:9092",
+                        "topics" : ["demo"],
+                        "consumer" : {
+                            "enabled": false
+                        },
+                        "producer": {
+                            "enabled": true
+                        }
+                    }
+                }
+            ]
+        }
+    ],
+    "flows": [
+        {
+            "name": "",
+            "selectors": [],
+            "request": [],
+            "response": [],
+            "subscribe": [],
+            "publish": [],
+            "enabled": true
+        }
+    ]
+}
+----
+
+== Quality of Service 
+
+You can set quality of service with the `qos` object in the `entrypoints` object, as shown in the following example. It can have a value of `none`, `auto`, `at-most-once`, or `at-least-once`.
```

Comment:
I'm not a big fan of copy-paste of the above descriptor as there is 100% chances that it will go out of sync (and probably wrong) in the next couple of weeks.

Reply:
> As mentioned, there is a compatibility matrice to display for QoS capabilities between entrypoint & endpoint

Where do I find the compatibility matrix?

Reply:
> I'm not a big fan of copy-paste of the above descriptor as there is 100% chances that it will go out of sync (and probably wrong) in the next couple of weeks.

Would it be better to say something like "Refer to the JSON definition of the API for the values that `qos` can have" and add a link to the JSON?

Reply:
Compatibility matrixes added and reference to the JSON definition added.